### PR TITLE
kakoune: add v2023.08.05

### PR DESCRIPTION
**Pull request by taliaferro**
Also added the compiler constraints for this version as [described in the README](https://github.com/mawww/kakoune#building).

### CODE REVIEW

File: var/spack/repos/builtin/packages/kakoune/package.py [hunk 1]
```diff
@@ -16,13 +16,18 @@ class Kakoune(MakefilePackage):
...
+    )
     version(
         "2021.11.08", sha256="aa30889d9da11331a243a8f40fe4f6a8619321b19217debac8f565e06eddb5f4"
     )
 
     depends_on("ncurses")
 
     conflicts("%gcc@:8", when="@2021.11.08", msg="GCC version must be at least 9.0!")
+    conflicts("%gcc@:10.3", when="@2023.08.05", msg="GCC version must be at least 10.3!")
+    conflicts("%llvm@:11", when="@2023.08.05", msg="Clang version must be at least 11!")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/kakoune/package.py]: Ranges include the upper bound:
```suggestion
    conflicts("%gcc@:10.2", when="@2023.08.05", msg="GCC version must be at least 10.3!")
    conflicts("%llvm@:10", when="@2023.08.05", msg="Clang version must be at least 11!")
```

**Review by alalazo — CHANGES_REQUESTED** [review]

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
Ping me in a few days if there's no review from the maintainer.

**Comment by taliaferro** [comment]
@alalazo -- I haven't heard back from the maintainer in a while; is there anything else you'd suggest here?

**Comment by alalazo** [comment]
@taliaferro Nope, thanks for the PR!

**Pull request merged**